### PR TITLE
TW-1621: improve forward screen to dialog

### PR DESCRIPTION
**Pull request by Te-Z**
### Issue: 
https://github.com/linagora/twake-on-matrix/issues/1621

By the way, there was a bug when going back from forward screen to media viewer screen on mobile. It was because of a PopScope used for Android only.

#### Web
https://github.com/linagora/twake-on-matrix/assets/31937920/03eb694a-7fed-4d4d-9b8a-b769fec1320c

#### Web reponsive
https://github.com/linagora/twake-on-matrix/assets/31937920/0b2358ee-da0e-4129-8dc4-4377b6ec062e

#### Mobile (android)

https://github.com/linagora/twake-on-matrix/assets/31937920/b35c16d3-dd0e-4ace-aa41-057ae7fd7ec6

#### Mobile (android) forward from other app

https://github.com/linagora/twake-on-matrix/assets/31937920/8ccc9bd3-863b-4805-a613-f0555464d5ed

### CODE REVIEW

**Comment by sherlockvn** [comment]
i think its should bigger in height of dialog

**Comment by Te-Z** [comment]
> i think its should bigger in height of dialog

Size updated:

image

File: lib/presentation/mixins/media_viewer_app_bar_mixin/media_viewer_app_bar_mixin.dart [hunk 1]
```diff
@@ -29,17 +31,44 @@ mixin MediaViewerAppBarMixin {
...
 
+  Future<PopResultFromForward?> _showForwardMobileDialog(
+    BuildContext context,
+  ) async =>
+      await showDialog(
+        context: context,
+        useSafeArea: false,
+        useRootNavigator: false,
+        builder: (c) => const Forward(),
+      );
```
Comment (hoangdat) [inline lib/presentation/mixins/media_viewer_app_bar_mixin/media_viewer_app_bar_mixin.dart:53]: - any impaction if it is dialog in mobile? why not keep as before?
- please demo also the case `share file from other app to twake chat`

**Comment by hoangdat** [comment]
I can not click on `Forward to...` to get focus
image

**Comment by hoangdat** [comment]
corner pink background for every item must be remove

Screenshot 2024-04-08 at 13 28 59

File: lib/pages/forward/forward_view.dart [hunk 2]
```diff
@@ -35,38 +34,29 @@ class ForwardView extends StatelessWidget {
           textEditingController: controller.searchTextEditingController,
         ),
       ),
-      body: PopScope(
-        canPop: false,
-        onPopInvoked: (didPop) async {
-          if (PlatformInfos.isAndroid) {
-            controller.popScreen();
-          }
-        },
```
Comment (hoangdat) [inline lib/pages/forward/forward_view.dart:44]: how about the case swift to back?

File: lib/presentation/mixins/media_viewer_app_bar_mixin/media_viewer_app_bar_mixin.dart (hunk 1 above)
Comment (Te-Z) [reply lib/presentation/mixins/media_viewer_app_bar_mixin/media_viewer_app_bar_mixin.dart]: I've not seen any impaction on mobile. It is basically exactly the same as before but in a method to handle separately the mobile and web case line 36.

The videos have been updated in this PR's description. There's a demo of the case `share file from other app to twake chat`

File: lib/pages/forward/forward_view.dart (hunk 2 above)
Comment (Te-Z) [reply lib/pages/forward/forward_view.dart:44]: works as expected. Please see the videos in description

**Comment by Te-Z** [comment]
> > corner pink background for every item must be remove
> > Screenshot 2024-04-08 at 13 28 59
> 
> Any update for this case? @Te-Z

Yes has demontrated in the videos.
Before:
Capture d’écran du 2024-04-08 11-09-03

After:
Capture d’écran du 2024-04-08 11-13-42

**Comment by Te-Z** [comment]
> > I can not click on `Forward to...` to get focus image
> 
> Still error

Maybe you tried on the last deployment where the error was. Try again @nqhhdev , or see the videos where I test this case. I hope it will be ok :)

File: lib/pages/forward/forward_view.dart [hunk 3]
```diff
@@ -27,59 +27,172 @@ class ForwardView extends StatelessWidget {
...
+
+class _WebActionsButton extends StatelessWidget {
+  final ValueNotifier<String> selectedChatNotifier;
+
+  final ValueNotifier<Either<Failure, Success>?> forwardMessageNotifier;
+
+  final void Function() forwardAction;
+
+  const _WebActionsButton({
+    // ignore: unused_element
```
Comment (sherlockvn) [inline lib/pages/forward/forward_view.dart:103]: if it's unused you can remove it

File: lib/pages/forward/forward_view.dart [hunk 4]
```diff
@@ -27,59 +27,172 @@ class ForwardView extends StatelessWidget {
...
+
+  @override
+  Widget build(BuildContext context) {
+    return Padding(
+      padding: ForwardViewStyle.webActionsButtonPadding,
+      child: Row(
+        mainAxisAlignment: MainAxisAlignment.end,
+        children: [
+          TwakeTextButton(
+            onTap: () => context.pop(),
```
Comment (sherlockvn) [inline lib/pages/forward/forward_view.dart:118]: context.pop() may not work in web

File: lib/pages/forward/forward_view.dart [hunk 5]
```diff
@@ -27,59 +27,172 @@ class ForwardView extends StatelessWidget {
...
       ),
     );
   }
 }
 
-class ForwardButton extends StatelessWidget {
-  const ForwardButton({
+class _ForwardButton extends StatelessWidget {
+  const _ForwardButton({
+    // ignore: unused_element
```
Comment (sherlockvn) [inline lib/pages/forward/forward_view.dart:195]: same

File: lib/pages/forward/forward_view.dart (hunk 4 above)
Comment (Te-Z) [reply lib/pages/forward/forward_view.dart:118]: It works as expected:

[Capture vidéo du 09-04-2024 09:48:03.webm](https://github.com/linagora/twake-on-matrix/assets/31937920/ad209020-8786-485a-8e8d-e23bdaa41d4f)

**Comment by sherlockvn** [comment]
> > > I can not click on `Forward to...` to get focus image
> > 
> > 
> > Still error
> 
> Maybe you tried on the last deployment where the error was. Try again @nqhhdev , or see the videos where I test this case. I hope it will be ok :)

Yeah, i tested it for huy, and it works fine. But there is another problem. When loading forward, you should disable cancel button or x button to prevent unexpected behavior

**Comment by Te-Z** [comment]
@sherlockvn X button updated to disappear when loading forward. I added widget tests also

**Review by sherlockvn — APPROVED** [review]

**Review by hoangdat — APPROVED** [review]

**Comment by hoangdat** [comment]
please rebase

**Pull request merged**